### PR TITLE
fix(cli/napi): handle all property variants in napi_define_properties

### PR DESCRIPTION
**Pull request by littledivy**
Fixes https://github.com/denoland/deno/issues/17509

This fixes the bug that blocked loading `fsevents` in Deno.

### CODE REVIEW

File: test_napi/src/properties.rs [hunk 1]
```diff
@@ -36,45 +53,56 @@ pub fn init(env: napi_env, exports: napi_value) {
...
       getter: None,
       setter: None,
       data: ptr::null_mut(),
       attributes: enumerable | writable,
-      value: number,
+      value: init_constants(env),
     },
     napi_property_descriptor {
-      utf8name: "test_property_r\0".as_ptr() as *const c_char,
+      utf8name: "test_property_rw\0".as_ptr() as *const c_char,
```
Comment (bartlomieju) [inline test_napi/src/properties.rs:66]: You can use `cstr!()` helper here

File: test_napi/src/properties.rs [hunk 2]
```diff
@@ -36,45 +53,56 @@ pub fn init(env: napi_env, exports: napi_value) {
...
+      utf8name: "test_property_r\0".as_ptr() as *const c_char,
+      name: ptr::null_mut(),
       method: None,
       getter: None,
       setter: None,
       data: ptr::null_mut(),
       attributes: enumerable,
       value: number,
     },
+    // TODO(@littledivy): Fix this.
```
[the remark this answers is not in the document]
Comment (littledivy) [reply test_napi/src/properties.rs:85]: `name` field isn't handled correctly. https://github.com/denoland/deno/issues/17712

File: test_napi/src/properties.rs (hunk 1 above)
Comment (littledivy) [reply test_napi/src/properties.rs:66]: I think it's overkill to use std's `CString`. `as_ptr` is obvious for a known comptime string. `cstr!` also allocates for the last NULL byte

Comment (bartlomieju) [reply test_napi/src/properties.rs:66]: This is testing code, I believe readability here is more important than allocating one allocation.

Comment (littledivy) [reply test_napi/src/properties.rs:66]: ok ill do it but this makes no difference; just that now ive to know about this new macro.

```rs
cstr!("test_property_rw")
// vs
"test_property_rw\0".as_ptr()
```

Comment (bartlomieju) [reply test_napi/src/properties.rs:66]: You don't have to do `as *const c_char` cast as well 😋

**Review by bartlomieju — APPROVED** [review]
LGTM

File: test_napi/src/lib.rs [hunk 3]
```diff
@@ -27,7 +27,7 @@ pub mod typedarray;
 #[macro_export]
 macro_rules! cstr {
   ($s: literal) => {{
-    std::ffi::CString::new($s).unwrap().as_ptr()
+    std::ffi::CString::new($s).unwrap().into_raw()
```
Comment (littledivy) [inline test_napi/src/lib.rs:30]: the allocate cstring wasn't leaked so the pointer from `.as_ptr` was dangling. `into_raw` leaks it

Comment (bartlomieju) [reply test_napi/src/lib.rs:30]: Good catch!

**Pull request merged**